src/actions/closeVault.ts

```diff
@@ -0,0 +1,120 @@
+import { Connection } from '../Connection';
+import { Wallet } from '../wallet';
+
+import { ActivateVault, CombineVault, Vault, VaultProgram } from '../programs/vault';
+import { Keypair, PublicKey } from '@solana/web3.js';
+import { AccountLayout, Token, TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import { CreateTokenAccount, Transaction } from '../programs';
+import { sendTransaction } from '.';
+import BN from 'bn.js';
+import { TransactionsBatch } from '../utils/transactions-batch';
+
+interface CloseVaultParams {
+  connection: Connection;
+  wallet: Wallet;
+  vault: PublicKey;
+  fractionMint: PublicKey;
+  fractionTreasury: PublicKey;
+  redeemTreasury: PublicKey;
+  priceMint: PublicKey;
+  externalPriceAccount: PublicKey;
```

Comment:
can we retrieve these params from vault?

Reply:
I don't think so

Reply:
https://github.com/metaplex-foundation/metaplex/blob/9d5a5c6d668cd9c597cff8c63dfba00dee2f72f0/js/packages/web/src/actions/unwindVault.ts#L52